kolibri_explore_plugin/assets/src/modules/topicsRoot/handlers.js

```diff
@@ -181,7 +181,21 @@ export function showFilteredChannels(store) {
   );
 }
 
-export function searchChannels(store, search, kind) {
+export function searchChannelsOnce(store, search, kind) {
+  const searchPromise = searchChannels(store, search, kind);
```

Comment:
This doesn't solve the problem of the `searchChannels` function updating the `topicsRoot/SET_SEARCH_RESULT`. This is the same as before, but done in the `handlers.js` instead of in the `SearchPage` component, but doesn't avoid the update of the store with search results for old searches.

This implementation will mitigate one of the problems in the `SearchPage` component, the one of duplicate rows of the same kind in the results, but won't fix the issue with content from a previous search appearing in the current one, mixed with the last search.